Http/YotpoRetry.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace Yotpo\Core\Http;
+
+use Magento\Framework\DataObject;
+use Yotpo\Core\Model\Config;
+use Magento\Framework\Exception\NoSuchEntityException;
+
+class YotpoRetry
+{
+    /**
+     * @var Config
+     */
+    protected $config;
+
+    const IS_SUCCESS_KEY = 'is_success';
+    const STATUS_KEY = 'status';
+
+    /**
+     * Processor constructor.
+     * @param Config $config
+     */
+    public function __construct(Config $config) {
+        $this->config = $config;
+    }
+
+    /**
+     * Yotpo retry requests mechanism
+     *
+     * @return DataObject
+     */
+    public function retryRequest($request)
```

Comment:
I'm not sure that `retryRequest` is the proper name here. I would expect that it'll get the closure after trying once, cause it **re**try.
What about `executeReuqestWithRetry`?

Reply:
Discussed offline, agreed `executeRequest`.

Reply:
🍭

Reply:
🔧